### DISCUSSION TITLE
miopenTransformTensor doesn't support data conversion

### DISCUSSION THREAD

**Discussion by densamoilov** [ROCmSoftwarePlatform/MIOpen#1704] [Q&A]
`miopenTransformTensor` only allows to change layout of data without data type conversion. There are generally 3 cases where data conversion is used:
1. Quantization (low precision inference): f32 -> int8(x4) and f16->int8(x4) conversion (with applying a scale)
2. Downconvert (low precision inference): f32 -> f16
3. Upconvert: f16 -> f32 (for operations that don't support f16)

`cudnnTransformTensor` supports all the 3 cases. Can you please suggest how to migrate from cuDNN to MIOpen without loosing the functionality descried above?

**Comment by densamoilov** [comment]
@JehandadKhan, @junliume, friendly ping.

**Comment by junliume** [comment]
@JehandadKhan  @carlushuang could you help with the comment?

**Comment by JehandadKhan** [comment]
Hello @densamoilov, Thanks for  drawing our attention to this disparity. We have added this to our roadmap and will update the issue when we have a PR to fix this.

**Comment by densamoilov** [comment]
Thank you !

**Comment by densamoilov** [comment]
@JehandadKhan, can you please suggest a way to perform the data conversion? Is there something in HIP, MIOpen, rocBLAS, etc that can help to do the conversion while waiting for it to be enabled in `miopenTransformTensor`?

**Comment by atamazov** [comment]
@densamoilov Can you please indicate the importance of the cases (1) (2) (3) from the topmost comment for you? I hope we'll be able to begin working on this soon; there is related ticket #2828 that we are going cope with.

/cc @junliume @JehandadKhan for usual prioritization and assignment.

**Reply by densamoilov** [reply]
The (1) and (2) are the most important ones for us as they are necessary to have an efficient low precision pipeline. The (3) is a lower priority for us.
